### ISSUE TITLE
[return-await] False positive on any return type, e.g. return JSON.parse()

### ISSUE THREAD

**Issue by felixfbecker** [typescript-eslint/typescript-eslint#1269]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/return-await": "error"
  }
}
```

```TS
async function fetcher(): Promise<string> {
    const response = await fetch('/.api/graphql')
    const responseBody = await response.text()
    try {
        return JSON.parse(responseBody)
    } catch (error) {
        return responseBody
    }
}
```

**Expected Result**

No error, as `JSON.parse()` does not return a Promise.

**Actual Result**

Error: returning an awaited promise is required in this context

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.9.0` |
| `@typescript-eslint/parser`        | `2.9.0` |
| `TypeScript`                       | `3.7.2` |
| `ESLint`                           | `6.6.0` |
| `node`                             | `12` |

**Comment by bradzacher** [comment]
Hmm, I missed this in the PR, but it looks like the rule was built this way to treat `any` and `unknown` as awaitable.

This shouldn't be the case (as per the decisions made for `promise-function-async`).
